### PR TITLE
feat: update iOS agent to native

### PR DESCRIPTION
**Pull request by ilfa**

### CODE REVIEW

File: example/ios/Podfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Uncomment this line to define a global platform for your project
```
Comment (makma) [inline example/ios/Podfile:1]: This comment doesn't make sense anymore, does it?

**Comment by makma** [comment]
Shouldn't be [link](https://github.com/fingerprintjs/fingerprintjs-pro-flutter/blob/6e5ba596150283778b5130dbf6a207b4b3f0de05/README.md?plain=1#L35) `Fingerprint Pro` instead (since it's a name of the lib since v2)?

**Review by makma — DISMISSED** [review]

File: ios/Classes/SwiftFpjsProPlugin.swift [hunk 2]
```diff
@@ -14,38 +14,57 @@ public class SwiftFpjsProPlugin: NSObject, FlutterPlugin {
...
-                initFpjs(token: token, endpoint: endpoint, region: region)
+                initFpjs(token: token, region: region)
                 result("Successfully initialized FingerprintJS Pro Client")
             } else {
                 result(FlutterError.init(code: "errorApiToken", message: "missing API Token", details: nil))
             }
         } else if (call.method == "getVisitorId") {
             let args = call.arguments as? Dictionary<String, Any>
-            let tags = args?["tags"] as? FingerprintJSProClient.Tags
+            let tags = args?["tags"] as? [String: JSONTypeConvertible]
```
Comment (petrpalata) [inline ios/Classes/SwiftFpjsProPlugin.swift:27]: I wonder whether this type conversion should really fail silently. If the developer passes anything that doesn't convert to `JSONTypeConvertible`, the entire `tags` dictionary is thrown away without any indication.

File: ios/Classes/SwiftFpjsProPlugin.swift [hunk 3]
```diff
@@ -14,38 +14,57 @@ public class SwiftFpjsProPlugin: NSObject, FlutterPlugin {
...
     }
 
-    private func getVisitorId(tags: FingerprintJSProClient.Tags?, _ handler: @escaping (String) -> Void) {
-        fpjsClient?.getVisitorId { result in
-            switch result {
-            case let .failure(error):
-                handler(error.localizedDescription)
-            case let .success(visitorId):
-                handler(visitorId)
+    private func getVisitorId(_ metadata: Metadata?, _ result: @escaping FlutterResult) {
```
Comment (petrpalata) [inline ios/Classes/SwiftFpjsProPlugin.swift:61]: The method never calls the closure when `fpjsClient` is `nil`. It was the case before but we should probably do something about it (`guard let client = fpjsClient else { result(<failure>) }` maybe?).

**Review by petrpalata — APPROVED** [review]

**Pull request merged**